src/commons/api/rrdp.rs

```diff
@@ -313,6 +316,29 @@ impl CurrentObjects {
         }
     }
 
+    /// Returns a copy of self where elements matching the given URI
+    /// are removed if there are any matches. Otherwise, returns None.
+    pub fn purge(&self, uri: &uri::Rsync) -> Option<Self> {
+        let mut withdraws = vec![];
+
+        // We first loop through the elements to avoid having to clone in case there is no work
+        for (hash, el) in &self.0 {
+            if el.uri() == uri || (uri.as_str().ends_with('/') && el.uri().as_str().starts_with(uri.as_str())) {
```

Comment:
This same check is also done by the caller `purge_files()`.... maybe this fn should just purge an exact set that it's given? Or the caller should use the result of this call rather than work out the matching set itself as well?

Reply:
The context is different. First we remove the matching files from the repo content.  But.. the objects are *also* kept on a per publisher basis and we will need to delete them there to. Presumably most publishers will have nothing to delete. So this is done this way to avoid cloning unless it's really needed - i.e. we found a publisher for which we need to do work: make an event with a copy of the (to be) deleted objects.